detox/e2e/test/account/push_notification_settings.e2e.ts

```diff
@@ -0,0 +1,111 @@
+        // * Verify basic elements on push notification settings screen
+        await expect(PushNotificationSettingsScreen.backButton).toBeVisible();
+        await expect(PushNotificationSettingsScreen.allNewMessagesOption).toBeVisible();
+        await expect(PushNotificationSettingsScreen.mentionsOnlyOption).toBeVisible();
+        await expect(PushNotificationSettingsScreen.nothingOption).toBeVisible();
+        await expect(PushNotificationSettingsScreen.pushThreadsFollowingOptionToggledOn).toBeVisible();
+        await expect(PushNotificationSettingsScreen.mobileOnlineOption).toBeVisible();
+        await expect(PushNotificationSettingsScreen.mobileAwayOption).toBeVisible();
+        await expect(PushNotificationSettingsScreen.mobileOfflineOption).toBeVisible();
+    });
+
+    it('MM-T5108_2 - should be able to change push notification settings and save by tapping navigation back button', async () => {
+        // # Tap on all new messages option, tap on mobile online option, tap on back button, and go back to push notification settings screen
+        await PushNotificationSettingsScreen.allNewMessagesOption.tap();
+        await PushNotificationSettingsScreen.mobileOnlineOption.tap();
+        await PushNotificationSettingsScreen.back();
+        await PushNotificationSettingsScreen.open();
+
+        // * Verify all new messages option and mobile online option are selected and push threads following option does not exist
+        await expect(PushNotificationSettingsScreen.allNewMessagesOptionSelected).toBeVisible();
+        await expect(PushNotificationSettingsScreen.mobileOnlineOptionSelected).toBeVisible();
+        await expect(PushNotificationSettingsScreen.pushThreadsFollowingOptionToggledOn).not.toExist();
+
+        // # Tap on nothing option, tap on back button, and go back to push notification settings screen
+        await PushNotificationSettingsScreen.nothingOption.tap();
+        await PushNotificationSettingsScreen.back();
+        await PushNotificationSettingsScreen.open();
+
+        // * Verify nothing option is selected and mobile options do no exist
+        await expect(PushNotificationSettingsScreen.nothingOptionSelected).toBeVisible();
+        await expect(PushNotificationSettingsScreen.mobileOnlineOption).not.toExist();
+        await expect(PushNotificationSettingsScreen.mobileAwayOption).not.toExist();
+        await expect(PushNotificationSettingsScreen.mobileOfflineOption).not.toExist();
+
+        // # Tap on mentions only option, tap on mobile away option, toggle push threads following option off, tap on back button, and go back to push notification settings screen
+        await PushNotificationSettingsScreen.mentionsOnlyOption.tap();
+        await PushNotificationSettingsScreen.mobileAwayOption.tap();
+        await PushNotificationSettingsScreen.togglePushThreadsFollowingOptionOff();
+        await PushNotificationSettingsScreen.back();
+        await PushNotificationSettingsScreen.open();
```

Comment:
Wonder if we can test not only the settings being shown, but that they work as expected (i.e. you get the correct notifications when something happens).

Reply:
In-app notifications are the only ones available in detox. Notifications are a tricky area since v1 in terms of e2e